### ISSUE TITLE
[gatsby-plugin-catch-links] - Missing resources cause links to not work

### ISSUE THREAD

**Issue by liamness** [gatsbyjs/gatsby#13903]
## Description

When resources fail to fetch, causing the initial hydration of the app to fail, gatsby-plugin-catch-links continues to call `navigate()` despite nothing being able to handle this. This results in broken links which cannot be used to navigate.

### Steps to reproduce

- Create a project using gatsby-plugin-catch-links, with a page linking to another without using `<Link />` but instead using `<a />`
- Build the site and then remove / rename / block some resource (e.g. anything `component---path-to-component-<hash>.js` seems to work).
- Visit a page which uses this resource and try to navigate from one page to another by clicking the aforementioned `<a />` element.

### Expected result

Despite the app failing to hydrate, the browser should be able to navigate from one page to another, falling back on the default non-SPA behaviour.

### Actual result

The click event is swallowed and the default behaviour prevented. No navigation occurs.

### Environment

```
  System:
    OS: Linux 4.15 Ubuntu 18.04.2 LTS (Bionic Beaver)
    CPU: x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
    Shell: 5.4.2 - /usr/bin/zsh
  Binaries:
    Node: 10.15.3 - /tmp/yarn--1557232001187-0.10878563852082901/node
    Yarn: 1.15.2 - /tmp/yarn--1557232001187-0.10878563852082901/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v10.15.3/bin/npm
  Languages:
    Python: 2.7.15 - /usr/bin/python
  Browsers:
    Chrome: 73.0.3683.103
    Firefox: 66.0.3
  npmPackages:
    gatsby: ^2.1.18 => 2.1.18 
    gatsby-plugin-catch-links: ^2.0.5 => 2.0.5 
    gatsby-plugin-google-analytics: ^2.0.10 => 2.0.10 
    gatsby-plugin-less: ^2.0.11 => 2.0.11 
    gatsby-plugin-react-helmet: ^3.0.0 => 3.0.0 
    gatsby-plugin-sharp: ^2.0.7 => 2.0.7 
    gatsby-plugin-styled-components: ^3.0.0 => 3.0.0 
    gatsby-remark-autolink-headers: ^2.0.7 => 2.0.7 
    gatsby-remark-copy-linked-files: ^2.0.5 => 2.0.5 
    gatsby-remark-images: ^2.0.4 => 2.0.4 
    gatsby-remark-katex: ^3.0.1 => 3.0.1 
    gatsby-remark-prismjs: ^3.0.2 => 3.0.2 
    gatsby-remark-relative-images: ^0.2.0 => 0.2.0 
    gatsby-remark-responsive-iframe: ^2.0.5 => 2.0.5 
    gatsby-source-filesystem: ^2.0.5 => 2.0.5 
    gatsby-transformer-remark: ^2.1.8 => 2.1.8 
```

**Comment by wardpeet** [comment]
Do you have like a real life use case where this happend? I can reproduce but I think I miss how you can reproduce it with a real life example.

**Comment by liamness** [comment]
I've come across this due to an unfortunately unsteady hosting and deployment (FTP-based) situation that is out of my control to change. I understand that for a lot of people, with rock-steady off the shelf solutions like Netlify, this is a solved problem! If I can, I will get us to migrate to such a solution. But it could also happen if you have a poor wi-fi or mobile signal, I suppose. I'm thinking it could happen in such a way:

Someone on a train clicks on a link to your site -> train goes under a tunnel -> the intial HTML loads but some resources don't -> site renders correctly but "caught" links do not work

**Comment by liamness** [comment]
Oh and just in case you didn't notice, I have opened a PR to fix: gatsbyjs/gatsby#13904

**Comment by wardpeet** [comment]
I did notice, thanks for the reminder though. 

> But it could also happen if you have a poor wi-fi or mobile signal, I suppose. I'm thinking it could happen in such a way.

this makes a lot of sense indeed.
